backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/orchestration/impl/ElasticsearchBatchingWorkOrchestrator.java

```diff
@@ -107,8 +108,9 @@ protected void doStart(ConfigurationPropertySource propertySource) {
 	}
 
 	@Override
-	protected void doSubmit(ElasticsearchBatchedWork<?> work, OperationSubmitter operationSubmitter) throws InterruptedException {
-		executors.get( work.getQueuingKey() ).submit( work, operationSubmitter );
+	protected void doSubmit(ElasticsearchBatchedWork<?> work, OperationSubmitter operationSubmitter,
+			Function<ElasticsearchBatchedWork<?>, Runnable> blockingRetryProducer) throws InterruptedException {
+		executors.get( work.getQueuingKey() ).submit( work, operationSubmitter, blockingRetryProducer );
```

Comment:
I pushed a commit to add generics, so that we could remove the unnecessary lambda whose only purpose was to cast a work.
Please have a look at that commit and let's discuss it here if you disagree.